internal/host/plugin/repository.go

```diff
@@ -24,7 +25,7 @@ type Repository struct {
 // only be used for one transaction and it is not safe for concurrent go
 // routines to access it. WithLimit option is used as a repo wide default
 // limit applied to all ListX methods.
-func NewRepository(r db.Reader, w db.Writer, kms *kms.Kms, opt ...Option) (*Repository, error) {
+func NewRepository(r db.Reader, w db.Writer, kms *kms.Kms, opt ...host.Option) (*Repository, error) {
```

Comment:
IMO it's a little strange to put the options 1 level up in the package hierarchy, and only for a portion of the methods on this repo.  This is especially true since this plugin package's option still has withLimit option.  We'll just need to break them out again if Lookup/List behavior diverges.  Is the intent to be able to build a shared interface used by authorize session?

Reply:
Mainly my intent was to be able to pass options from the service handlers, and we have a lot of places where logic is duplicated (good catch on the limit option, forgot to remove). My thinking was that if we did want to pass an actual type specific option we could simply wrap it within a host option that chains into a call into the typed object.

Reply:
We just want to be careful to not add dependencies for packages `internal/A/B` and `internal/A/C` where code in A depends on code in either `B` or `C`.

Reply:
Yeah -- I figure the more specific code would rely on the less specific code, but not the other way around. We can iterate on this as we go.

Reply:
Removed duplicate option; closing this for now.